pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go

```diff
@@ -0,0 +1,354 @@
+	if err != nil {
+		var statusErr error
+		var syncFailed string
+		if upstreamConfig == nil {
+			syncFailed = ": syncing failed"
+		}
+		cluster = cluster.DeepCopy()
+		apimgmtv3.ClusterConditionUpdated.False(cluster)
+		apimgmtv3.ClusterConditionUpdated.Message(cluster, fmt.Sprintf("[Syncing error%s] %s", syncFailed, err.Error()))
+
+		cluster, statusErr = c.clusterClient.Update(cluster)
+		if statusErr != nil {
+			return cluster, statusErr
+		}
+	} else if strings.Contains(apimgmtv3.ClusterConditionUpdated.GetMessage(cluster), "[Syncing error") {
+		cluster = cluster.DeepCopy()
+		apimgmtv3.ClusterConditionUpdated.True(cluster)
+		apimgmtv3.ClusterConditionUpdated.Message(cluster, "")
+		cluster, err = c.clusterClient.Update(cluster)
+		if err != nil {
+			return cluster, err
+		}
+	}
+	if upstreamConfig == nil {
+		return cluster, err
+	}
+
+	switch {
+	case cluster.Spec.AKSConfig != nil:
+		if !reflect.DeepEqual(cluster.Status.AKSStatus.UpstreamSpec, upstreamConfig.aksConfig) {
+			logrus.Infof("updating cluster [%s], upstream change detected", cluster.Name)
+			cluster = cluster.DeepCopy()
+			cluster.Status.AKSStatus.UpstreamSpec = upstreamConfig.aksConfig
+			cluster, err = c.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, err
+			}
+		}
+
+		if !reflect.DeepEqual(cluster.Spec.AKSConfig, cluster.Status.AKSStatus.UpstreamSpec) {
+			logrus.Infof("cluster [%s] currently updating, skipping spec sync", cluster.Name)
+			return cluster, nil
+		}
+
+		// check for changes between EKS spec on cluster and the EKS spec on the EKSClusterConfig object
+		specMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(cluster.Spec.AKSConfig)
+		if err != nil {
+			return cluster, err
+		}
+
+		upstreamSpecMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(upstreamConfig.aksConfig)
+		if err != nil {
+			return cluster, err
+		}
+
+		var updateConfig bool
+		for key, value := range upstreamSpecMap {
+			if specMap[key] == nil {
+				continue
+			}
+			if reflect.DeepEqual(specMap[key], value) {
+				continue
+			}
+			updateConfig = true
+			specMap[key] = value
+		}
+
+		if !updateConfig {
+			logrus.Infof("cluster [%s] matches upstream, skipping spec sync", cluster.Name)
+			return cluster, nil
+		}
+
+		if err = runtime.DefaultUnstructuredConverter.FromUnstructured(specMap, cluster.Spec.AKSConfig); err != nil {
+			return cluster, err
+		}
+	case cluster.Spec.EKSConfig != nil:
+		if !reflect.DeepEqual(cluster.Status.EKSStatus.UpstreamSpec, upstreamConfig.eksConfig) {
+			logrus.Infof("updating cluster [%s], upstream change detected", cluster.Name)
+			cluster = cluster.DeepCopy()
+			cluster.Status.EKSStatus.UpstreamSpec = upstreamConfig.eksConfig
+			cluster, err = c.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, err
+			}
+		}
+
+		if !reflect.DeepEqual(cluster.Spec.EKSConfig, cluster.Status.EKSStatus.UpstreamSpec) {
```

Comment:
This should be `!reflect.DeepEqual(cluster.Spec.EKSConfig, cluster.Status.AppliedSpec.EKSConfig)`.

Reply:
I fixed it